engine/common/oncall_gateway/client.py

```diff
@@ -0,0 +1,155 @@
+from dataclasses import dataclass, field
+from json import JSONDecodeError
+from typing import List, Optional
+from urllib.parse import urljoin
+
+import requests
+
+SERVICE_TYPE_ONCALL = "oncall"
+
+
+@dataclass
+class SlackLink:
+    service_type: str
+    service_tenant_id: str
+    slack_team_id: str
+
+
+@dataclass
+class MSTeamsLink:
+    service_type: str
+    service_tenant_id: str
+    msteams_id: str
+
+
+@dataclass
+class Tenant:
+    service_tenant_id: str
+    service_type: str
+    cluster_slug: str
+    slack_links: List[SlackLink] = field(default_factory=list)
+    msteams_links: List[MSTeamsLink] = field(default_factory=list)
+
+
+class ChatopsProxyAPIException(Exception):
+    """A generic 400 or 500 level exception from the Chatops Proxy API"""
+
+    def __init__(self, status, url, msg="", method="GET"):
+        self.url = url
+        self.status = status
+        self.method = method
+
+        # Error-message returned by chatops-proxy.
+        # Since chatops-proxy is internal service messages shouldn't be exposed to the user
+        self.msg = msg
+
+    def __str__(self):
+        return f"LabelsRepoAPIException: status={self.status} url={self.url} method={self.method} error={self.msg}"
+
+
+class ChatopsProxyAPIClient:
+    def __init__(self, url: str, token: str):
+        self.base_url = url
```

Comment:
nit: looks like `self.base_url` is only used in the constructor, consider removing it